x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/allocation/SearchableSnapshotDiskThresholdIntegTests.java

```diff
@@ -268,6 +275,73 @@ public void testHighWatermarkCanBeExceededOnColdOrFrozenNode() throws Exception
         });
     }
 
+    public void testOvercommitInitializingSearchableSnapshotShards() throws Exception {
+        internalCluster().startMasterOnlyNode();
+        internalCluster().startNode(
+            Settings.builder()
+                .putList(NodeRoleSettings.NODE_ROLES_SETTING.getKey(), DiscoveryNodeRole.DATA_HOT_NODE_ROLE.roleName())
+                .build()
+        );
+
+        InternalClusterInfoService masterInfoService = (InternalClusterInfoService) internalCluster().getCurrentMasterNodeInstance(
+            ClusterInfoService.class
+        );
+        ClusterInfoServiceUtils.refresh(masterInfoService);
+
+        int nbIndices = createIndices();
+
+        assertAcked(
+            client().admin()
+                .cluster()
+                .preparePutRepository("repository")
+                .setType("mock")
+                .setSettings(Settings.builder().put("location", randomRepoPath()).build())
+        );
+        var mockRepository = (MockRepository) internalCluster().getCurrentMasterNodeInstance(RepositoriesService.class)
+            .repository("repository");
+        // Prevent searchable snapshot shards from quickly jumping from INITIALIZED to STARTED
+        mockRepository.setBlockOnceOnReadSnapshotInfoIfAlreadyBlocked();
```

Comment:
Let's move this before we mount the snapshot, to avoid possible interactions while we take the snapshot.

Reply:
I believe we already do. The `createSnapshot("repository", "snapshot", nbIndices)` line is after the `setBlockOnceOnReadSnapshotInfoIfAlreadyBlocked` call

Reply:
I meant to move it before we call `mountIndices(indicesStoresSizes, "mounted-", "repository", "snapshot", FULL_COPY);`

Reply:
Moved